org.eclipse.scanning.points/src/org/eclipse/scanning/jython/JythonInterpreterManager.java

```diff
@@ -0,0 +1,240 @@
+package org.eclipse.scanning.jython;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Properties;
+
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.scanning.points.ScanPointGeneratorFactory;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.wiring.BundleWiring;
+import org.python.core.Py;
+import org.python.core.PyString;
+import org.python.core.PySystemState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * An interpreter manager which is designed to create an interpreter
+ * which either is sympathic and working with the GDA configuration or
+ * creates an empty and working interpreter.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+class JythonInterpreterManager {
+
+	private static final Logger logger = LoggerFactory.getLogger(JythonInterpreterManager.class);
+	
+	
+	private static volatile PySystemState configuredState;
+
+	/**
+	 * Call to ensure that an interpreter is set up and configured and
+	 * able to load the relevant bundles.
+	 */
+	public static synchronized void setupSystemState(String... bundleNames) {
+		
+		ClassLoader loader=null;
+		if (configuredState==null) { // Relies on setupSystemState() being called early in the server startup.
+			loader = createJythonClassLoader(PySystemState.class.getClassLoader());
+	 		initializePythonPath(loader); 
+		}
+		
+    	PySystemState state = Py.getSystemState();
+    	if (state==configuredState) return;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)